File: atcoder/abc/go/abc219/abc219_C/abc219_C.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"os"
	"sort"
	"strconv"
)

type scanner struct { sc *bufio.Scanner }
func newScanner(input io.Reader) *scanner {
	sc := bufio.NewScanner(input)
	sc.Split(bufio.ScanWords)
	sc.Buffer(make([]byte, 1024), int(1e+9))
	return &scanner{sc}	
}
var rdr = newScanner(os.Stdin)
const BUFSIZE = 10000000
var wrtr = bufio.NewWriterSize(os.Stdout, BUFSIZE)
func gs() string  { rdr.sc.Scan(); return rdr.sc.Text() }
func gi() int     { i,e := strconv.Atoi(gs()); if e != nil {panic(e)}; return i }

func translate(s string, key []byte) string {
	ans := make([]byte,len(s))
	for i,c := range s { ans[i] = key[int(c)] }
	return string(ans)
}

func main() {
	//f1, _ := os.Create("cpu.prof"); pprof.StartCPUProfile(f1); defer pprof.StopCPUProfile()
    defer wrtr.Flush()
	infn := ""
	if infn == "" && len(os.Args) > 1 {	infn = os.Args[1] }
    if infn != "" {	f, e := os.Open(infn); if e != nil { panic(e) }; rdr = newScanner(f) }
	X := gs()
	N := gi()
	S := make([]string,N)
	for i:=0;i<N;i++ { S[i] = gs() }
	S2 := make([]string,N)
	key1 := make([]byte,256)
	key2 := make([]byte,256)
	for i,c := range X { key1[int(c)] = 'a'+byte(i); key2[int('a')+i] = byte(c) }
	for i:=0;i<N;i++ { S2[i] = translate(S[i],key1) }
	sort.Slice(S2,func(i,j int)bool{return S2[i]<S2[j]})
	S3 := make([]string,N)
	for i:=0;i<N;i++ { S3[i] = translate(S2[i],key2) }
	for _,s := range S3 { fmt.Fprintln(wrtr,s) }

}

File: abc/python/abc219/abc219_C.py

import sys
infile = sys.stdin.buffer

def gs()  : return infile.readline().rstrip()
def gi()  : return int(gs())
def gf()  : return float(gs())
def gss() : return gs().split()
def gis() : return [int(x) for x in gss()]
def gfs() : return [float(x) for x in gss()]

def main(infn="") :
    global infile
    infile = open(infn,"r") if infn else open(sys.argv[1],"r") if len(sys.argv) > 1 else sys.stdin
    X = gs()
    N = gi()
    S = []
    for _ in range(N) : S.append(gs())
    t1 = {}; t2 = {}
    for c,d in zip(X,"abcdefghijklmnopqrstuvwxyz") :
        t1[ord(c)] = ord(d)
        t2[ord(d)] = ord(c)
    for i in range(N) : S[i] = S[i].translate(t1)
    S.sort()
    for i in range(N) : S[i] = S[i].translate(t2)
    for s in S : print(s)

if __name__ == '__main__' :
    main()
    sys.stdout.flush()

